### PR TITLE
Adding ability to specify an HTTP timeout

### PR DESCRIPTION
**Pull request by zyphlar**
Also changing Guzzle link in readme to point to appropriate version, and mentioning how to obtain test data.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -314,7 +314,7 @@ If you are using PHP 5.3 with an autoloader besides Composer, you must load
 ## Contributing ##
 
 Patches and pull requests are encouraged. All code should follow the
-PSR-2 style guidelines. Please include unit tests whenever possible.
+PSR-2 style guidelines. Please include unit tests whenever possible. You may obtain the test data for the maxmind-db folder from https://github.com/maxmind/MaxMind-DB
```
Comment (oschwald) [inline README.md]: These are actually available as a submodule in this repo. You can retrieve them with `git submodule update --init --recursive` or adding a `--recursive` to your initial clone.

File: src/GeoIp2/WebService/Client.php [hunk 2]
```diff
@@ -62,13 +64,17 @@ class Client implements ProviderInterface
...
      */
     public function __construct(
         $userId,
         $licenseKey,
         $locales = array('en'),
         $host = 'geoip.maxmind.com',
-        $guzzleClient = null
+        $guzzleClient = null,
+        $timeout = null,
+        $connectTimeout = null
```
Comment (oschwald) [inline src/GeoIp2/WebService/Client.php]: I wish we would have used an options array here as this is getting a bit unmanageable, but I guess that is water under the bridge.

File: src/GeoIp2/WebService/Client.php [hunk 3]
```diff
@@ -180,7 +188,14 @@ private function responseFor($endpoint, $class, $ipAddress)
 
         $client = $this->guzzleClient ?
             $this->guzzleClient : new GuzzleClient();
-        $request = $client->get($uri, array('Accept' => 'application/json'));
+        $options = array();
+        if($this->timeout != null){
```
Comment (oschwald) [inline src/GeoIp2/WebService/Client.php]: Could you make this and the next one `!==`.

File: src/GeoIp2/WebService/Client.php [hunk 4]
```diff
@@ -62,13 +64,17 @@ class Client implements ProviderInterface
      * @param string $host Optional host parameter
      * @param object $guzzleClient Optional Guzzle client to use (to facilitate
      * unit testing).
+     * @param string $timeout Total transaction timeout
+     * @param string $connectTimeout Initial connection timeout
```
Comment (oschwald) [inline src/GeoIp2/WebService/Client.php]: Can you also specify that they are in seconds?

File: README.md (hunk 1 above)
Comment (zyphlar) [inline README.md]: Thanks! Added this to the readme.

File: src/GeoIp2/WebService/Client.php (hunk 2 above)
Comment (zyphlar) [inline src/GeoIp2/WebService/Client.php]: That's what I was thinking. I didn't want to break anything, just add the possibility. "Slow is the new broken," as they say.

File: src/GeoIp2/WebService/Client.php (hunk 3 above)
Comment (zyphlar) [inline src/GeoIp2/WebService/Client.php]: Done, good catch. Though specifying a timeout of 0 or false seems logically equivalent to me :confused:

File: src/GeoIp2/WebService/Client.php (hunk 4 above)
Comment (zyphlar) [inline src/GeoIp2/WebService/Client.php]: Done

File: src/GeoIp2/WebService/Client.php (hunk 3 above)
Comment (oschwald) [inline src/GeoIp2/WebService/Client.php]: Yeah, I agree that they are generally the same, but I want it to be clear that we have the same behavior as Guzzle when it is set to `0`.

**Comment by oschwald** [comment]
Merged. Thanks!

**Comment by zyphlar** [comment]
No, thank you!

**Pull request merged**